assets/js/dpulc_viewer.jsx

```diff
@@ -1,20 +1,93 @@
 import Viewer from "@samvera/clover-iiif/viewer";
 import React from 'react';
 import Loader from './loader';
+import { createStitches } from '@stitches/react';
+const { styled } = createStitches({
+  media: {
+    sm: '(min-width: 640px)',
+    md: '(min-width: 768px)',
+    lg: '(min-width: 1024px)',
+    xl: '(min-width: 1280px)'
+  },
+});
 // DpulcViewer is a react component which acts as a wrapper for Clover with
 // all of our default settings and functionality.
-const handleCanvasIdCallback = (activeCanvasId, pushEvent) => {
+let currentCanvas = null
+const handleCanvasIdCallback = (activeCanvasId, loadedCanvasIdx, pushEvent) => {
+  // In vertical thumbnails, scroll to view.
+  scrollThumbnail(activeCanvasId, loadedCanvasIdx - 1)
   // Tell LiveView that we've changed the canvas so we can change the URL or
   // anything else, if necessary.
   if(typeof pushEvent === 'function') {
     pushEvent("changedCanvas", { "canvas_id": activeCanvasId })
   }
 };
+
+const scrollThumbnail = (activeCanvasId, loadedCanvasIdx) => {
```

Comment:
How does this only impact vertical thumbnails? Is it the isOutsideVertically logic?

Can you add a comment to this function about why we need it for vertical thumbnails and not for horizontal? I assume there is similar code for the horizontal layout in clover itself?

Reply:
@hackartisan Good idea! Added.